### ISSUE TITLE
com.xiwang.csmall.product.filter

### ISSUE THREAD

**Issue by xufree888** [zXiWang/csmall-pp#1]
当有多个模块时候都需要重写一次这个过滤器么?

**Comment by zXiWang** [comment]
你可以试试用微服务

**Comment by xufree888** [comment]
但是如果是后台业务, 一般只部署在一台内部服务器上, 这个时候不同的服务使用RPC通信,是不是无形的增加了失败的风险, 而且RPC毕竟是网络通信, 是不是对性能也有影响? 还要采用分布式事务等操作, 增加了维护成本, 另外采用微服务是不是会部署起来很麻烦?

**Comment by zXiWang** [comment]
是的,没有完美的方案,既然想统一管理那就得付出其他代价.微服务部署起来相较而言麻烦点,但后期维护会好很多,当然是对于模块多改动多而言.没这种大需求每个需要的模块粘一份filter就好了



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年11月29日(星期二) 下午2:40
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [zXiWang/csmall-pp] com.xiwang.csmall.product.filter (Issue #1)





 
但是如果是后台业务, 一般只部署在一台内部服务器上, 这个时候不同的服务使用RPC通信,是不是无形的增加了失败的风险, 而且RPC毕竟是网络通信, 是不是对性能也有影响? 还要采用分布式事务等操作, 增加了维护成本, 另外采用微服务是不是会部署起来很麻烦?

**Comment by xufree888** [comment]
非常感谢您的解答, 解决了我看到这段教程以来一直有的一个疑虑!

**Comment by zXiWang** [comment]
没事,互相学习




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年11月30日(星期三) 凌晨1:08
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [zXiWang/csmall-pp] com.xiwang.csmall.product.filter (Issue #1)





 
非常感谢您的解答, 解决了我看到这段教程以来一直有的一个疑虑!

**Comment by xufree888** [comment]
> 没事,互相学习
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: ***@***.***&gt;; 发送时间: 2022年11月30日(星期三) 凌晨1:08 收件人: ***@***.***&gt;; 抄送: ***@***.***&gt;; ***@***.***&gt;; 主题: Re: [zXiWang/csmall-pp] com.xiwang.csmall.product.filter (Issue #1) 非常感谢您的解答, 解决了我看到这段教程以来一直有的一个疑虑! — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

您看是否方便添加一下您的微信或者QQ方便今后技术上的交流 ?

**Comment by zXiWang** [comment]
不好意思,我也不是专业的,有什么问题直接在github上问就好了:)

**Comment by xufree888** [comment]
我看你酷鲨商城微服务版本的在不同的模块也都独立的添加了[security](https://github.com/zXiWang/CoolShark-SpringCloud/tree/master/csmall-repo/mall-order/mall-order-webapi/src/main/java/cn/tedu/mall/order/security)

**Comment by zXiWang** [comment]
这是学习阶段的一种暴力的过滤方法,就是在所有模块都加一份,不是真的微服务思想,真正微服务思想是spring cloud里的网关,比如里面有一个gateway模块,它可以统一配置管理所有模块过滤器达到前面一样的效果

**Comment by zXiWang** [comment]
这是学习阶段的一种暴力的过滤方法,就是在所有模块都加一份,不是真的微服务思想,真正微服务思想是spring cloud里的网关,比如里面有一个gateway模块,它可以统一配置管理所有模块过滤器达到前面一样的效果

**Comment by zXiWang** [comment]
这是网关的优缺点,复制的
**1.** 网关可以屏蔽后端服务的协议细节，比如有的服务可能采用的rpc方式，有的服务采用的restful方式，有的甚至可能采用的比较古老的web Service，在网关这一层我们可以统一通信协议，这样客户端调用起来会方便很多。
**2.** 可以统一整个系统的认证和授权功能，甚至可以统一不同的客户端采用的不同认证方式，比如Cookie认证，Session认证等。可以统一设置后端服务的保护策略，比如：不同的服务可以设置不同的白名单系统，不同的服务可以采用不同的限流策略等。
**3.** 在网关这个统一的入口，更容易做日志的收集工作。

虽然网关很香，但是一旦整个系统引入网关，就会随之带来一些问题：

**1.** 增加了网关，在请求的整个过程中就至少多了一层转发，性能会有稍微下降。
**2.** 网关如果出现问题，降造成整个系统不稳定，甚至down机，这就是网关的单点问题。所以一般网关都会有多个节点。

**Comment by xufree888** [comment]
有在网关中实现的Demo么 ? 我去参考学习一下!
